File: Recursion/Nokia.py
# cook your dish here
# cook your dish here


def findmin(n,m,d,pos,sum=0,index= 0):
    if index==n:
        left = findleft(n,d,pos)
        right = findleft(n,d,pos)
        sum = sum + left + right
        #print(m-sum)
        return m-sum 
    if(pos>0):
        d[pos]= 1 #only one or zero that's it.
        left = findleft(n,d,pos)
        right = findright(n,d,pos)
        sum += left + right
    lst =[]
    for key,value in d.items():
        if key>0 and key <n+1 and value==0:
            lst.append(findmin(n,m,d,key,sum,index+1))
    lst = list(filter(lambda x:x!=-1,lst))
        
    #lst = [findmin(m,d,pos+i,sum,index+1) for i in range(1,3-index+1)]  
    
        #a=findmin(m,d,pos+1,sum,index+1)
       #b=findmin(m,d,pos+2,sum,index+1)
       #c=fidmin(m,d,pos+3,sum,index+1)
    val = min(lst) if len(lst)!=0 else -1
    d[pos]=0
    return val
    
    
#pos 3 means third level in the recursion 3 , meaning 2 pos are filld.
def findleft(n,d,pos):
    if pos==1:
        return 1
    else:
        pos = pos-1
        k=0
        while pos>=0 and d[pos]==0:
            k=k+1
            pos=pos-1
        
        return k
        
    
        
def findright(n,d,pos):
    if pos==n:
        return 1
    else:
        k=0
        for i in range(pos+1,n+2):
            if d[pos]==0:
                k=k+1
            else:
                break
        return k
        
        

t = int(input())
while t>0:
    inputs = list(map(int,input().split()))
    n = inputs[0]
    m = inputs[1]
    pos=0
    d={}
    for  i in range(0,n+2):
        d[i]=0
    cost = findmin(n,m,d,pos,0,0)
    print(cost)
    t=t-1

#k,n-k+1   
#spot 1 to n
#pos=1 (the solderies order)

    
        
        
        
    
    
    
    
    
    

File: Recursion/nokia.cpp
#include <iostream>
using namespace std;

int findleft(int n , int d[], int pos){
   if(pos==1){
       return 1;
   }
   
   int k=0;
   for(int i = pos-1; i>=0;i--){
       
       k = k+1;
       if(d[i]==1){
           return k;
       }
   }
    
    
}


int findright(int n, int d[], int pos){
    
    if(pos==n){
        return 1;
    } 
    
    int k=0;
    for(int i=pos+1;i<=n+1;i++){
        k=k+1;
        if(d[i]==1){
            return k;
        }
    }
    
}


int findmin(int n,int m,int d[],int pos,int sum=0,int index= 0){
    if(index==n){
        sum += 2;
        return m-sum>=0?m-sum:-1;
    }
    
    if(pos>0){
       d[pos]=1;
       int left = findleft(n,d,pos);
       int right = findright(n,d,pos);
       sum += left+right;
    }
    
    int min = 9999;
    int cost =0;
    for(int i=1;i<=n;i++){
        if(d[i]==0){
         cost = findmin(n,m,d,i,sum,index+1);
         if(cost >=0 && cost<min){
            min = cost;
         }
        }
        
    }
    d[pos]=0;
    return min!=9999?min :-1;
      
    }

int main() {
	// your code goes here
	int t;
	int n,m;
	int pos=0;
	cin>>t;
	while(t>0){
	    
	    cin>> n >> m;
	    int d[n+2]={0};
	    int cost = findmin(n,m,d,pos,0,0);
	    cout<<cost<<endl;
	    --t;
	}
	return 0;
}
